system/jlib/jlog.cpp

```diff
@@ -977,6 +755,13 @@ void CategoryLogMsgFilter::reset()
 
 // HandleLogMsgHandler
 
+void HandleLogMsgHandlerTable::handleMessage(const LogMsg & msg)
+{
+    CriticalBlock block(crit);
+    msg.toStringTable(curMsgText.clear(), messageFields);
```

Comment:
question: would it be more efficient to build this string on the stack, before entering the crit? (but would require that there was a variety of StringBuffer that did not take ownership of passed in buffer)

Reply:
It is hard to build on the stack because there isn't an upper limit.  This reuses a member variable to avoid reallocation.

However I don't think more than one call to handleMessage can ever occur at the same time.  If so the crit is there to protect access to handle - so the formatting could probably be moved outside the critsec.